linodecli/plugins/metadata.py

```diff
@@ -0,0 +1,217 @@
+"""
+This plugin allows users to access the metadata service while in a Linode.
+
+Usage:
+
+   linode-cli metadata [ENDPOINT]
+"""
+
+import argparse
+import sys
+
+from linode_metadata import MetadataClient
+from linode_metadata.objects.error import ApiError
+from linode_metadata.objects.instance import ResponseBase
+from requests import ConnectTimeout
+from rich import print as rprint
+from rich.table import Table
+
+PLUGIN_BASE = "linode-cli metadata"
+
+
+def process_sub_columns(subcolumn: ResponseBase, table: Table, values_row):
+    """
+    Helper method to process embedded ResponseBase objects
+    """
+    for key, value in vars(subcolumn).items():
+        if isinstance(value, ResponseBase):
+            process_sub_columns(value, table, values_row)
+        else:
+            table.add_column(key)
+            values_row.append(str(value))
+
+
+def print_instance_table(data):
+    """
+    Prints the table that contains information about the current instance
+    """
+    attributes = vars(data)
+    values_row = []
+
+    table = Table()
+
+    for key, value in attributes.items():
+        if isinstance(value, ResponseBase):
+            process_sub_columns(value, table, values_row)
+        else:
+            table.add_column(key)
+            values_row.append(str(value))
+
+    table.add_row(*values_row)
+    rprint(table)
+
+
+def print_ssh_keys_table(data):
+    """
+    Prints the table that contains information about the SSH keys for the current instance
+    """
+    table = Table(show_lines=True)
+
+    table.add_column("ssh keys")
+    for key in data.users.root:
```

Comment:
It looks like `root` will be None if no authorized keys are configured for the root user, so we'll need a None check here to prevent an error.

Side note: It looks like this endpoint will also return keys for users not configured by the API, so we may need to update py/go-metadata accordingly 👍